app/assets/javascripts/Components/__tests__/submission_file_upload_modal.test.jsx

```diff
@@ -0,0 +1,34 @@
+import {shallow} from "enzyme";
+import SubmissionFileUploadModal from "../Modals/submission_file_upload_modal";
+
+describe("For the SubmissionFileUploadModal", () => {
+  it("checks if only selection menu renders", () => {
```

Comment:
These test descriptions should include an English description of the state involved, not just what you're checking.

Moreover, there are two dimensions of state that you should be considering: the `onlyRequiredFiles`/`requiredFiles` props values, *and* how many files are actually being uploaded, i.e., are in the state (0/1/2+). This is a good time to use nested `describe` blocks, each with their own `beforeEach` that you can use to set up common test state across multiple tests.